packages/core/src/decorators/oas3/remove-unused-components.ts

```diff
@@ -2,17 +2,28 @@ import { isEmptyObject } from '../../utils';
 
 import type { Location } from '../../ref-utils';
 import type { Oas3Decorator } from '../../visitors';
-import type { Oas3Components, Oas3Definition } from '../../typings/openapi';
+import type {
+  Oas3Definition,
+  Oas3_1Definition,
+  Oas3ComponentsBase,
+  Oas3_1Components,
+  Oas3Schema,
+  Oas3_1Schema,
+} from '../../typings/openapi';
 
 export const RemoveUnusedComponents: Oas3Decorator = () => {
   const components = new Map<
     string,
-    { usedIn: Location[]; componentType?: keyof Oas3Components; name: string }
+    {
+      usedIn: Location[];
+      componentType?: keyof (Oas3ComponentsBase<Oas3Schema | Oas3_1Schema> | Oas3_1Components);
```

Comment:
Related to https://github.com/Redocly/redocly-cli/pull/1795/files?diff=split&w=1#r1898410733 :
```suggestion
      componentType?: keyof (Oas3Components | Oas3_1Components);
```

Reply:
Did you actually update it?